From 0a8d7a7a4627a8e6c6c16e7c61e7c7d477510855 Mon Sep 17 00:00:00 2001
From: "Yue, VincentX" <vincentx.yue@intel.com>
Date: Fri, 23 Mar 2018 15:09:37 +0800
Subject: [PATCH] Fix the Message alaways reply to the first sender throung
 notification

According to PendingIntent description, if you want to
replace its extra data with what is in this new Intent,
you must add the flag FLAG_UPDATE_CURRENT and add a new
requestCode to indicate it is a intent.

Change-Id: I187438b181e730705479977738e3444df1ce127f
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-59676
Signed-off-by: Yue, VincentX <vincentx.yue@intel.com>
---
 src/com/android/car/messenger/MapMessageMonitor.java | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/com/android/car/messenger/MapMessageMonitor.java b/src/com/android/car/messenger/MapMessageMonitor.java
index b07d263..a0b5ab7 100644
--- a/src/com/android/car/messenger/MapMessageMonitor.java
+++ b/src/com/android/car/messenger/MapMessageMonitor.java
@@ -229,7 +229,7 @@ class MapMessageMonitor {
                         }
                         PendingIntent LaunchPlayMessageActivityIntent = PendingIntent.getActivity(
                                 mContext,
-                                REQUEST_CODE_VOICE_PLATE,
+                                notificationInfo.mNotificationIdForReply,
                                 getPlayMessageIntent(senderKey, notificationInfo),
                                 0);
 
@@ -253,7 +253,7 @@ class MapMessageMonitor {
                                     .setSound(Settings.System.DEFAULT_NOTIFICATION_URI);
                         }
                         mNotificationManager.notify(
-                                notificationInfo.mNotificationId, builder.build());
+                                notificationInfo.mNotificationIdForReply, builder.build());
                     }
                 });
     }
@@ -291,7 +291,7 @@ class MapMessageMonitor {
             Intent replyIntent = getPlayMessageIntent(senderKey, notificationInfo);
             replyIntent.putExtra(PlayMessageActivity.EXTRA_SHOW_REPLY_LIST_FLAG, true);
             PendingIntent autoReplyIntent = PendingIntent.getActivity(
-                    mContext, REQUEST_CODE_AUTO_REPLY, replyIntent, 0);
+                    mContext, notificationInfo.mNotificationIdForReply, replyIntent, PendingIntent.FLAG_UPDATE_CURRENT);
             builders.add(new Notification.Action.Builder(icon,
                     mContext.getString(R.string.action_reply), autoReplyIntent));
         }
@@ -322,7 +322,7 @@ class MapMessageMonitor {
                 .setAction(action)
                 .putExtra(MessengerService.EXTRA_SENDER_KEY, senderKey);
         return PendingIntent.getService(mContext,
-                notificationInfo.mNotificationId, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                notificationInfo.mNotificationIdForMute, intent, PendingIntent.FLAG_UPDATE_CURRENT);
     }
 
     void clearNotificationState(SenderKey senderKey) {
@@ -436,7 +436,7 @@ class MapMessageMonitor {
         while (notificationIt.hasNext()) {
             Map.Entry<SenderKey, NotificationInfo> entry = notificationIt.next();
             if (predicate.test(entry.getKey())) {
-                mNotificationManager.cancel(entry.getValue().mNotificationId);
+                mNotificationManager.cancel(entry.getValue().mNotificationIdForReply);
                 notificationIt.remove();
             }
         }
@@ -631,7 +631,8 @@ class MapMessageMonitor {
     private static class NotificationInfo {
         private static int NEXT_NOTIFICATION_ID = 0;
 
-        final int mNotificationId = NEXT_NOTIFICATION_ID++;
+        final int mNotificationIdForReply = NEXT_NOTIFICATION_ID++;
+        final int mNotificationIdForMute = NEXT_NOTIFICATION_ID++;
         final String mSenderName;
         @Nullable
         final String mSenderContactUri;
-- 
1.9.1

